src/core/config/group_by.rs

```diff
@@ -1,34 +1,98 @@
+use std::fmt::Display;
+
 use serde::{Deserialize, Serialize};
 
 use crate::core::is_default;
 #[derive(Clone, Debug, Eq, Serialize, Deserialize, PartialEq, schemars::JsonSchema)]
 /// The `groupBy` parameter groups multiple data requests into a single call. For more details please refer out [n + 1 guide](https://tailcall.run/docs/guides/n+1#solving-using-batching).
 pub struct GroupBy {
     #[serde(default, skip_serializing_if = "is_default")]
-    path: Vec<String>,
```

Comment:
Let's store `path` and `key` here 

```rust
struct GroupBy {
   path: Vec<String>,
   key: Option<String>
}
```

If the `key` is empty then we should use the last value in the path. Otherwise the use can specify a key specifically.